sympy/physics/vector/point.py

```diff
@@ -444,6 +444,9 @@ def v1pt_theory(self, otherpoint, outframe, interframe):
         _check_frame(outframe)
         _check_frame(interframe)
         self._check_point(otherpoint)
+        if not otherpoint.vel(interframe) == 0:
```

Comment:
Given my comment above, this should be something like "if otherpoint velocity is defined and it isn't zero in the shared frame, then ...."

Reply:
Yes, I agree. We could at most display a warning. The best option I see is to use `try` `except` to catch the `ValueError`, when the velocity is not defined. This will result in something like:
```py
other_point_is_valid = True
try:
    if not otherpoint.vel(interframe)==0:
        otherpoint_is_valid = False
except ValueError:
    pass
if not other_point_is_valid:
    raise ValueError(...)
```
Advantage is that this is a extensive test, which really checks whether the velocity is defined. Disadvantages: it uses `try` `except` and will possibly add `{interframe: 0}` to the `otherpoint._vel_dict' by calling `otherpoint.vel` (not sure if this is a disadvantage).

Shall I implement this solution? Or is there a better one? Also should I revert my previous commit, so the test files and `model.py` will not be changed?

Reply:
It is fine to do explicit checks over the try/except clause. Maybe preferred in this case.

We try to avoid warnings in SymPy, but we already have some for this particular piece of code. I think it is fine to throw an error if the velocity is defined and it is not constant. If it is not defined, the behavior should be as it is, don't do anything.

Reply:
Should I also undo my previous changes to the other files?

Reply:
Yes, unfortunately. I don't think we want to add this requirement that you have to set the velocity of the base point in the shared frame. It is really unnecessary. The alternative is, as @angadhn suggested, to set that velocity automatically for the user. It's probably ok to do that. So, if the velocity is undefined, set it to 0 and if it is defined and not zero, then throw an error. The only thing I'm hung up on that is that we don't set more than one velocity with any other methods. But I may be warmed up to the idea by now. If it throws an error if it is not constant and sets it to zero otherwise, I can't see a major issue, but then we wouldn't return that second velocity.  But that method really shouldn't return a velocity, it should just set them. If you set the velocity to zero and make that very clear in the documentation, then that's probably fine.

Reply:
I will just revert my changes and implement the [suggestion](https://github.com/sympy/sympy/pull/23359#discussion_r850122669). For this one the velocity in that frame will only be set to `0` if it is already known that it is `0`.